### DISCUSSION TITLE
Markdown Diagram Support

### DISCUSSION THREAD

**Discussion by JasonDeland-pcc** [standardnotes/forum#2480] [Suggestions]
Could you add support for mermaid?

**Comment by atmoio** [comment]
Not familiar with it. Would it be a standalone editor, or integrated into one of the existing editors? Something like this for standalone? https://mermaidjs.github.io/mermaid-live-editor

Also, probably won't work on this until there is some more demand for it.

**Comment by JasonDeland-pcc** [comment]
Could it be integrated into the advanced preview for markdown?

**Comment by atmoio** [comment]
Not sure..depends on what open-source libraries there are.

**Comment by JasonDeland-pcc** [comment]
I would love to help under my personal githib account - jaydeland

**Comment by andaag** [comment]
This is something that's supported in atom/vscode markdown stuff as well, I think there are multiple open source implementations for rendering this.

Looking at how popular those extensions are might be an indicator of how interesting this is to have.

**Comment by darthShadow** [comment]
Would love to have this available. So, +1 for the request.

**Comment by sklum** [comment]
Also very interested in this as an extension.  Looks like it's been requested in #445 as well.

**Comment by dreamflasher** [comment]
I came here to create a new issue for mermaid, and am happy to see that others are interested, too! Would be a great addition.

**Comment by dreamflasher** [comment]
markdown-it, the editor which is used in https://github.com/sn-extensions/simple-markdown-editor has an extension for mermaid (https://www.npmjs.com/package/markdown-it-mermaid) so it should be relatively easy to be added there?

**Comment by atmoio** [comment]
It would seem like it, but the plugin seems to be abandoned.

**Comment by dreamflasher** [comment]
But the plugin is just a wrapper for mermaid, it's merely calling it, essentially it's finished?

**Comment by atmoio** [comment]
Hmm maybe. I don't currently have the capacity for this, but take a look here: https://github.com/standardnotes/forum/issues/839#issuecomment-578812983

We could do a similar arrangement for this if someone wants to submit a PR, for a $100 bounty.

**Comment by myreli** [comment]
The retired [Markdown Visual](https://github.com/standardnotes/plugins) has integrated mermaid support.

<img width="865" alt="image" src="https://user-images.githubusercontent.com/17554234/218532969-b7540391-10d1-412c-8193-f818bf0dd4aa.png">
